### PR TITLE
Improve project structure

### PR DESCRIPTION
**Pull request by Orviss**
Features:
* removed IDE specific files (nbproject, .classpath, etc.)
* moved files to correct maven structure (src/main/java)
* moved resource files to resources directory 

There are a few things that I would like to add -> mainly checkstyle plugin and errors on compilation warnings but I think it would be wiser to include them in later PRs.

Thanks for review :)

### CODE REVIEW

**Comment by idodeclare** [comment]
I like the idea of this, but please I do think it should be coordinated to happen at the most ideal time for active contributors who may have active, imminent branches — since this PR touches almost every file. E.g., I have 25 branches in line for review that touch ~450 files.

I didn't examine this PR closely, so I'll ask: will this make it difficult for new contributors to begin work? I.e., the current project recommends NetBeans and seems structured to support that IDE.

Also I think that probably it would be best if this restructuring is done at the same time as issue #1778 (retired "opensolaris" from naming), since that issue also will touch almost every file and is supported by @tarzanek and @vladak — and should be coordinated likewise.

(Lastly, I do not like to see tangential changes in a PR such as that a4febf6 commit you tacked on.)

**Comment by Orviss** [comment]
Hi,

first of all, I asked first on Slack, if you are so eager you should have replied there. Secondly, I discussed this with @tarzanek and @vladak who supported this change. 

This change will not make it hard for new contributors. Quite the opposite since Maven is IDE agnostic. Honestly, NetBeans dependency was a bad decision from the start.

Regarding the last commit, look at the previous ones. I had to add there method for creating temporary directory because of the cyclic dependencies between the modules. Alas, I forgot about it and created PR before it was repaired.

If you are so worried about some of those PRs then I will personally repair them if they break.

Nevertheless, I think you could have replied to the PR more politely and not in such a hotheaded way. We do not wage war but work together towards a common goal.

Yours sincerely,
Adam Hornáček

**Comment by idodeclare** [comment]
@Orviss , the only thing hotheaded is your response. My very first sentence included the word "please" out of politeness. Now contrast with your response where your first sentence includes "if you are so eager." Yes, I am eager, having contributed 74,979 ++/ 21,395 -- lines to OpenGrok, with many tens of thousands of lines of net contributions in queue.

You might have mentioned the Slack conversation calmly, and then I would have alerted the project that perhaps https://opengrok.slackarchive.io —which I monitor—is not working correctly. There are no updates there since January.

Regardless, it's absurd that you think that having missed the Slack conversation, my request for coordination in my replying to your PR is impolite. (Where do you even read "hotheaded" or hahaha "war" in my reply?)

The wording on your last commit makes no mention of your mistake — in fact the wording hides it. With a more honest wording — such as "Remove method accidentally left in" — I would not have called it out as tangential.

**Comment by vladak** [comment]
I bet we can shove both sets of changes in, it just needs to be coordinated. It seems to me that Chris is more worried about having to merge his queue of PRs than the actual changes in this PR if I understand this correctly.

**Comment by Orviss** [comment]
What I think will break are the newly created files. This will be my main branch for quite a while so I'm willing to merge any changes as they come so it won't be such a pain. 

Btw, I like the idea of renaming the packages as well. I can see that <http://opengrok.org> already works. May I include it into this branch as well?

What I propose is that we consider this change. We do not have to merge it right away. Active contributors can try this on their favourite IDEs and report caveats which bump into. (NetBeans will probably suffer a little since this is a generic way of project structure and not targeted right for it.) I can use this time to improve the Maven files (honestly, I did not work with submodules before and they do not work that well or sometimes as one would expect).

**Comment by tarzanek** [comment]
I will have a look why the archivebot in slack stopped working, sorry for that

**Comment by tarzanek** [comment]
If this PR will make the project work on every major IDE Out of Box(tm), then it'd be lovely ...
build.xml has historically a lot of convenience helpers, but I guess maven can help with most of them ...

File: jrcs/pom.xml [hunk 1]
```diff
@@ -1,19 +1,19 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
```
Comment (Orviss) [inline jrcs/pom.xml]: I do not like this much but here <https://maven.apache.org/developers/conventions/code.html> is written that `<project>` tag should be always on one line. However in these examples <https://maven.apache.org/pom.html> it is on multiple lines...

**Comment by vladak** [comment]
Yeah, the opengrok.org site will be eventually updated with the info about
the foundation (in progress).

Dne 6. 3. 2018 8:38 dop. napsal uživatel "Adam Hornáček" <
notifications@github.com>:

**Comment by vladak** [comment]
W.r.t. merging this let's go ahead and merge Chris'es PRs and then this
one. We can then fix the issues in other IDEs (assuming this will work
seamlesly in IDEA and Netbeans at least).

Dne 6. 3. 2018 8:38 dop. napsal uživatel "Adam Hornáček" <
notifications@github.com>:

**Comment by vladak** [comment]
I would like to merge this rather sooner than later. There seems to be a race between @idodeclare cranking out PR's and us doing reviews. Let's discuss a plan and schedule.

**Comment by tarzanek** [comment]
merge before release?

**Comment by vladak** [comment]
Yes, definitely wanted for 1.1.

Dne út 17. 4. 2018 22:58 uživatel Lubos Kosco <notifications@github.com>
napsal:

**Comment by tarzanek** [comment]
@idodeclare well to be completely honest - how confident are you in your changes? ;)
I am tempted to merge the rest 14 changes, but not sure how much regressions they will cause (and that's what worries me, since it will slow a release and we're already 1 year after last release ...)

**Comment by idodeclare** [comment]
> how confident are you in your changes

@tarzanek , very confident :) haha but more brains are better. The next one, "Add TokenizerMode; support `OPENGROK_ALL_NONWHITESPACE`" (dcbe903a1), is quite large so I'll be grateful for other eyes on it after #2066 is merged.

Later, "Add classes to support Objective-C analysis" (823e18f5a) is substantial because it introduces the possibility of simultaneous symbolic/literal representations of individual code locations and also handling Objective-C's syntax of interleaved method names and argument values with possible nesting of method calls. E.g.:

    initWithJSON:[method:abc forWhat:def] withAssetBundle:[NSBundle mainBundle]];

has two Objective-C argument-taking methods: `initWithJSON:withAssetBundle:` and `method:forWhat:`.

There is also "Refactor `AnalyzerGuru`, `HistoryGuru` as non-singleton" (47159d735e37) that touches a lot of files but is otherwise pretty straight-forward. The other eleven imminent PRs are of nominal size and also quite straight-forward — but again mostly order dependent.

**Comment by tarzanek** [comment]
let's merge this, I think we need to finally move on to clarify the build system

**Comment by vladak** [comment]
Yes, my plan is to merge this right after the suggester.

Dne čt 12. 7. 2018 7:59 uživatel Lubos Kosco <notifications@github.com>
napsal:

**Comment by tarzanek** [comment]
ah! so suggester still uses old layout?
I thought Adam builds on top of his mvn changes ...
but OK

**Comment by tarzanek** [comment]
btw. is @idodeclare OK with the merge?
Chris, will you be able to rebase?

**Comment by idodeclare** [comment]
@tarzanek yep OK with me

**Comment by vladak** [comment]
Ok, @Orviss needs to merge with himself first (the checkstyle changes) and then let's merge this PR.

**Comment by Orviss** [comment]
Problems:
* the configuration will need to be replaced with the new one because of the package renaming (when I tried the old configuration the suggester produced error that it could not find Lucene class -> no idea why, I can investigate if necessary)
* seemed that profile for Java 9/10 tests did not work -> specific arguments were not passed and ChronicleMap will not be loaded. Again, no idea why because I do not see anything marginally different.

**Comment by vladak** [comment]
The `Configuration` refresh is expected. Given this will be part of 1.1 the users will have to reindex from scratch anyway.

**Comment by vladak** [comment]
Please merge the conflicting files again.

**Comment by vladak** [comment]
Could you update the developer intro wiki ?

**Comment by Orviss** [comment]
> Could you update the developer intro wiki ?

Sure thing. I will try to do it this evening :)

**Pull request merged**